Sources/Fuzzilli/Modules/Storage.swift

```diff
@@ -66,6 +77,42 @@ public class Storage: Module {
             self.storeProgram(code, to: fileURL)
         }
 
+        fuzzer.registerEventListener(for: fuzzer.events.REPRLFail) { scripts in
+            assert(fuzzer.config.diagnostics == true)
+
+            let dirname = "/reprllog_\(String(currentMillis()))"
+            do {
+                try FileManager.default.createDirectory(atPath: self.REPRLFailDir + dirname, withIntermediateDirectories: true)
+            } catch {
+                self.logger.fatal("Failed to create storage directories. Is \(self.storageDir) writable by the current user?")
+            }
+
+            for (id, script) in scripts.enumerated() {
+                let filename = "/reprl_\(id).js"
+                // create the file URL
+                let fileURL = URL(fileURLWithPath: self.REPRLFailDir + dirname + filename)
+                self.storeProgram(script, to: fileURL)
+            }
+        }
+
+        fuzzer.registerEventListener(for: fuzzer.events.InvalidProgramFound) { program in
+            if fuzzer.config.diagnostics {
```

Comment:
Can you turn this around and only even register the listener if diagnostics mode is active?
I think you can just do 
```
if fuzzer.config.diagnostics {
    fuzzer.registerEventListener(for: x) {
    ....
    }

    fuzzer.registerEventListener(for: y) {
        ...
```
?